java-client-model/src/main/java/com/cumulocity/model/authentication/Jsons.java

```diff
@@ -23,7 +24,7 @@ private static Optional<String> readValueAsString(String field, String json) {
         if (StringUtils.isEmpty(field)) {
             return Optional.absent();
         }
-        Object value = JsonPath.read(json, "$." + field);
-        return value == null ? Optional.absent() : Optional.of(value.toString());
+        Map<?, ?> data = JSONParser.defaultJSONParser().parse(Map.class, json);
+        return Optional.fromNullable(data.get(field)).transform(String::valueOf);
```

Comment:
Do we know how much performance we need from this? @Michal-Bak marking you, since this helper method is used for authorization with O-Auth. I think sometimes 2-3 times for single request (tenantId, userId). 

Maybe we can rewrite it somehow (eg. using statefull class instead of those static methods) to not parse full json into same map several times. Unless we dont need tuned performance in this area. 

History in core repository says that initially this was done with Jackson, and then rewritten to use this jayway library.

Reply:
In core, also Jayway is used. Comparing this Svenson's usage and Jayway's performance should be very close. For sure in the future, it can be improved :), but I don't think this PR will change anything in how it works so far.